app/experimenter/static/rapid/components/forms/ExperimentForm.tsx

```diff
@@ -0,0 +1,170 @@
+import React from "react";
+import { useHistory } from "react-router-dom";
+
+interface ErrorListProperties {
+  errors: Array<string> | undefined;
+}
+
+const ErrorList: React.FC<ErrorListProperties> = ({ errors }) => {
+  if (!errors) {
+    return null;
+  }
+
+  return (
+    <ul className="text-danger pl-3 my-2">
+      {errors.map((e, i) => (
+        <li key={i}>{e}</li>
+      ))}
+    </ul>
+  );
+};
+
+const ExperimentForm: React.FC = () => {
+  const [formData, setFormData] = React.useState({
+    name: "",
+    objectives: "",
+  });
+  const [errors, setErrors] = React.useState({});
```

Comment:
based on your `ErrorList` component I think your default value here is `undefined`, no? I know this is WIP but it wouldn't hurt to define the types for your form and error states here more explicitly so they can be easily refactored (even though they are placeholder) i.e. 
```typescript
type ErrorList = Array<string> | undefined;
interface ErrorListProps {
  errors: ErrorList;
}
...
 const [errors, setErrors] = React.useState<ErrorList>();

```

Reply:
ill add some types but `errors` on this line is the errors for the form which is a map/object where each key (field name) maps to an array of strings (which are the error for that field). this is how django rest framework spits out errors.

the errorlist component lists errors for a single field and therefore either gets an array of strings if there were errors or undefined if there were no errors.

Reply:
Ah got it, so like `{[name: string]: ErrorList}`? Makes sense 👍